controllers/argocd/deployment.go

```diff
@@ -35,26 +35,30 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+var (
+	defaultRepoServerReplicas int32 = 1
+	defaultServerReplicas     int32 = 1
+)
+
 // getArgoCDRepoServerReplicas will return the size value for the argocd-repo-server replica count if it
-// has been set in argocd CR. Otherwise, nil is returned if the replicas is not set in the argocd CR or
+// has been set in argocd CR. Otherwise, default value is returned if the replicas is not set in the argocd CR or
```

Comment:
The reason we want to return nil as opposed to our default is to make a distinction that the user has explicit set the number of replicas. If the user does not set the value explicit, the code will honor the default value in the Depployment.